src/main/anovos/data_analyzer/association_evaluator.py

```diff
@@ -223,23 +217,14 @@ def variable_clustering(
 
     idf_sample = idf.sample(False, min(1.0, float(sample_size) / idf.count()), 0)
```

Comment:
As discussed, in here, can you replace `sample` with our `data_sample` function from anovos.data_ingest.data_sampling module. An example on how to implement it you can see in here:
https://github.com/anovos/anovos/blob/3d4c3d08ec4b868e182b4543831644a7a0ff2e3c/src/main/anovos/data_analyzer/association_evaluator.py#L109